src/js/index.js

```diff
@@ -0,0 +1,134 @@
+import { ALERT_MESSAGE, OPERATION, DIGIT_NUMBER_MEX_LENGTH } from './constants.js';
+import { $ } from './utils.js';
+
+function App() {
+  // 상태값 정의
+  this.state = {
+    total: '0',
+    digits: [],
+    calculateStep: 0,
+    operators: [],
+  };
+
+  // init 함수
+  this.init = () => {
+    initEventListeners();
+  };
+
+  // 렌더 함수
+  const render = () => {
+    $('#total').textContent = this.state.total;
+  };
+
+  // 계산식 정의
+  const operations = {
+    [OPERATION.PLUS]: (a, b) => Number(a) + Number(b),
+    [OPERATION.MINUS]: (a, b) => Number(a) - Number(b),
+    [OPERATION.MULTIPLICATION]: (a, b) => Number(a) * Number(b),
+    [OPERATION.DIVISION]: (a, b) => Number(a) / Number(b),
+  };
+
+  // 상태값 초기화 함수
+  const resetState = () => {
+    this.state.total = '0';
+    this.state.digits = [];
+    this.state.calculateStep = 0;
+    this.state.operators = [];
+  };
+
+  // 계산하는 함수
+  const calculate = () => {
+    // digit이 1개만 입력됐을경우
+    if (!this.state.digits[1]) {
+      this.state.operators[0] = '';
+      this.state.total = Math.floor(this.state.digits[0]);
+      render();
+      return;
+    }
+
+    const total = this.state.digits.reduce((prev, current, index) => {
+      if (prev !== 0 && this.state.operators[index - 1]) {
+        return operations[this.state.operators[index - 1]](prev, current);
+      }
+
+      return prev + Number(current);
+    }, 0);
+
+    resetState();
+    this.state.digits[0] = total + '';
+    this.state.total = Math.floor(total);
+    render();
+  };
+
+  // digit값 가져오는 함수
+  const getDigitNumber = (number, $digit) => {
+    if (!number) return $digit.textContent;
+    if (DIGIT_NUMBER_MEX_LENGTH - number.length < 1) {
+      alert(ALERT_MESSAGE.MAX_NUMBER);
+      return number;
+    }
+    return (number += $digit.textContent);
+  };
```

Comment:
```suggestion
  const getDigitNumber = (number, { textContent }) => {
    if (!number) return textContent;
    if (DIGIT_NUMBER_MEX_LENGTH - number.length < 1) {
      alert(ALERT_MESSAGE.MAX_NUMBER);
      return number;
    }
    return number + textContent;
  };
```
이렇게 표현할 수 있겠네요 ㅎㅎ
그리고 number에 값을 할당한다음에 반환하지 않아도 될 것 같아요!
어차피 매개변수거든요.

그리고 number가 지금 문자열 타입인 것 같은데, 변수 명은 number 라고 표현되어 있어서 다른 사람들이 보면 헷갈릴 수 있을 것 같아요!

Reply:
오 훨씬 보기 좋네요 감사하빈다!